### PR TITLE
James startup algo

### PR DESCRIPTION
**Pull request by JamesDoering**
added total_angle and total_angle_list.
Edited last_maxima and algos that use it to be more general.

### CODE REVIEW

File: Code/Single_Pendulum/single_increase_parametric.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def __init__(self, values, all_data, **kwargs):
         self.prev_time, self.curr_time = 0.0, values['time']
         self.moving_down = True
 
-        self.max_times = self.last_maxima(all_data)
+        self.max_times = self.last_maxima(all_data['time'])
```
Comment (GeorgeSheppard) [inline Code/Single_Pendulum/single_increase_parametric.py]: won't this break it? Think it is missing another list that needs to be passed in.

File: Code/utility_functions.py [hunk 2]
```diff
@@ -186,16 +186,24 @@ def last_zero_crossing(values, previous_time, previous_be):
...
-    # extracting a moving average of the previous encoder values to prevent incorrect maximas begin evaluated
-    # only use an odd number for window size, as indexes need to be adjusted below
-    avg_be = np.abs(moving_average(all_data['be'][-30:], window_size))
-    # extract index corresponding to latest maxima, index_max_angle(number of previous values to return)
-    angle_max_index = (np.diff(np.sign(np.diff(avg_be))) < 0).nonzero()[0] + 1 + (window_size - 1)/2
-    if be_time == 'time':
-        return all_data['time'][-30:][angle_max_index[-1]]
-    elif be_time == 'be':
-        return all_data['be'][-30:][angle_max_index[-1]]
+# def last_maxima(all_data, be_time='time', window_size=5):
```
Comment (GeorgeSheppard) [inline Code/utility_functions.py]: Can we just get rid of all the commented stuff if it's not needed anymore.

File: Code/utility_functions.py [hunk 3]
```diff
@@ -210,3 +218,20 @@ def last_minima(self, all_data):
...
 
     return min_times
+
+def total_angle(be, se0, se1):
+    """
+    Takes encoder values at a single time and converts to a total angle.
+    Used for triple pendulum examples.
+    """
+    x, y = position_seat_cartesian(be, se0, se1)
+    total_angle = np.arctan(x/y)
```
Comment (GeorgeSheppard) [inline Code/utility_functions.py]: do we want it returned in radians or degrees? I think degrees as this should be similar to the big encoder values.

Comment (JamesDoering) [reply Code/utility_functions.py]: all right big man I'll fix these later

File: Code/utility_functions.py [hunk 4]
```diff
@@ -210,3 +218,20 @@ def last_minima(self, all_data):
...
+    return total_angle
+
+def total_angle_list(all_data):
+    """returns the list of total_encoder values and matching times"""
+    total_angles = []
```
Comment (GeorgeSheppard) [inline Code/utility_functions.py:233]: I don't get what this does, you are looping over times but not using i or t in anything, so this just does that one line a lot of times? Also what is it used for?

**Review by GeorgeSheppard — CHANGES_REQUESTED** [review]
See comments for changes

**Pull request closed without merging**